be/src/exec/workgroup/scan_task_queue.cpp

```diff
@@ -66,23 +66,33 @@ void WorkGroupScanTaskQueue::close() {
 
     _is_closed = true;
     _cv.notify_all();
+    _cv_for_borrowed_cpus.notify_all();
 }
 
 StatusOr<ScanTask> WorkGroupScanTaskQueue::take() {
     std::unique_lock<std::mutex> lock(_global_mutex);
 
-    WorkGroupScanSchedEntity* wg_entity = nullptr;
-    while (wg_entity == nullptr) {
+    WorkGroupScanSchedEntity* wg_entity;
+    while (true) {
         if (_is_closed) {
             return Status::Cancelled("Shutdown");
         }
 
-        if (_wg_entities.empty()) {
-            _cv.wait(lock);
-            continue;
+        wg_entity = _take_next_wg();
```

Comment:
I do not understand why this function _take_next_wg is not changed,  it seems  executors of  a dedidcated resource group should always take wg_entity from its own resource group if borrowing flat is off? so why pick up a workgroup among several workgroups ?

Reply:
Yes, exclusive executors needn't this logic, and only shared/common need this logic. For convenience, I reused this code without making any changes.

Reply:
We can optimize this by not using the WorkGroupDriverQueue for exclusive executors and instead utilizing the regular QuerySharedDriverQueue. Let me handle this in the next PR.

Reply:
We can optimize this by replacing `WorkGroupDriverQueue` with the regular `QuerySharedDriverQueue` for exclusive executors. Let me check whether this is possible.

Reply:
Updated. Now, exclusive executors use non-workgroup task queue, while shared executors use workgroup task queue.

Reply:
so it is somewhat confusing,  can we use a new name, for an example,  take_next_task, take_next_entity. after introducing exclusive workgroup,  take_next_wg seems ambiguous.

Reply:
`take_next_wg ` is a little misunderstanding, since it will pick a `wg_entity` not a `task`.

I'd add some comments for this.